react-native/android/build.gradle

```
@@ -177,11 +178,35 @@ def findNdkBuildFullPath() {
         def ndkDir = System.getenv('ANDROID_NDK_HOME')
         return new File(ndkDir, getNdkBuildName()).getAbsolutePath()
     }
-    def ndkDir = android.hasProperty('plugin') ? android.plugin.ndkFolder :
-            plugins.getPlugin('com.android.library').sdkHandler.getNdkFolder()
-    if (ndkDir) {
-        return new File(ndkDir, getNdkBuildName()).getAbsolutePath()
+
+    def ndkDir = null
+
+    try {
+        // Reading [ndk.dir] from local.properties
+        Properties properties = new Properties()
+        properties.load(project.rootProject.file('local.properties').newDataInputStream())
+
+        ndkDir = properties.getProperty('ndk.dir')
+        if(ndkDir != null)
+            return new File(ndkDir, getNdkBuildName()).getAbsolutePath()
+    }catch(FileNotFoundException e){
+        println("[Warning]: local.properties not found ")
+        println("Error: ", e)
     }
+
+    try {
+
+        ndkDir = android.hasProperty('plugin') ? android.plugin.ndkFolder :
+                plugins.getPlugin('com.android.library').sdkHandler.getNdkFolder()
+
+        if (ndkDir) {
+            return new File(ndkDir, getNdkBuildName()).getAbsolutePath()
+        }
+    }catch(MissingPropertyException e){
```

What line would you expect throw this error?

```suggestion
    } catch(MissingPropertyException e) {
```

---

[This line ](https://github.com/realm/realm-js/blob/master/react-native/android/build.gradle#L181
) will throw a error like this: 

```
No such property: sdkHandler for class: com.android.build.gradle.LibraryPlugin
```